### PR TITLE
[AIRFLOW-6860] Default ignore_first_depends_on_past to True

### PR DESCRIPTION
**Pull request by zhongjiajie**
---
Issue link: [AIRFLOW-6860](https://issues.apache.org/jira/browse/AIRFLOW-6860)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by zhongjiajie** [comment]
ref https://github.com/apache/airflow/pull/7490#discussion_r386844032

**Comment by zhongjiajie** [comment]
CI sad with unrelated reason

```
tests/test_config_templates.py ....                                      [  0%]
tests/test_configuration.py ...................................          [  0%]
tests/test_core.py ........................                              [  1%]
tests/test_core_to_contrib.py .......................................... [  1%]
........................................................................ [  3%]
........................................................................ [  4%]
........................................................................ [  6%]
........................................................................ [  7%]
........................................................................ [  8%]
........................................................................ [ 10%]
........................................................................ [ 11%]
........................................................................ [ 12%]
........................................................................ [ 14%]
........................................................................ [ 15%]
........................................................................ [ 16%]
..........................................                               [ 17%]
tests/test_example_dags.py ..                                            [ 17%]
tests/test_example_dags_system.py .Timeout (0:08:00)!
Thread 0x00007f63a8dbf400 (most recent call first):
```

**Comment by zhongjiajie** [comment]
It's weird, it pass in my local breeze env, but failed in Travis both two of my PR. https://github.com/apache/airflow/pull/7593

**Comment by nuclearpinguin** [comment]
Yeah, sometimes the tests timeouts. I think @mik-laj was investigating it and we think is related to limited resources on CI (memory swap for example).

**Comment by zhongjiajie** [comment]
@nuclearpinguin Thanks for the clarification, so did it Travis side or ours?

**Comment by potiuk** [comment]
It looks like Travis quirks - it looks like the same tests start to fail more often and then after some time they continue to work more reliably. It' super-difficult trace (and know whose fault it is) and figure out what's going on. This is not a simple "We" vs. Travis. 

We already fixed (or workarounded) a number of problems with Travis and we have an outstanding task to migrate to Github Actions but we have other high priority tasks.

@zhongjiajie  - Maybe you could help with investigating it or migrating to Github Actions (where we believe things will be more stable)?

**Comment by zhongjiajie** [comment]
@potiuk Like to help, but I don't have much experience in CI/CD. I could see we already have some discuss in dev-list, and does the related AIP is? https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-23+Migrate+out+of+Travis+CI

**Comment by potiuk** [comment]
Yep. That's the related PR indeed :)

**Comment by zhongjiajie** [comment]
@potiuk @nuclearpinguin PTAL, thx.

**Comment by ashb** [comment]
Since the original change/PR hasn't been included in a release yet, this should be targeted against the same JIRA

**Comment by zhongjiajie** [comment]
> Since the original change/PR hasn't been included in a release yet, this should be targeted against the same JIRA

Got it

**Comment by zhongjiajie** [comment]
I use JIRA ticket, change PR title and git commit message

**Comment by ashb** [comment]
@zhongjiajie The reason for targetting this against AIRFLOW-6976 and not a new jira is around release-automation: if we have separate Jiras for a "related" fix then it is possible to miss this change if the first PR was backported. By having two commits target a single jira issue our existing release scripts (`./dev/airflow-jira`) will show both commits need to be pulled in.

**Comment by zhongjiajie** [comment]
> @zhongjiajie The reason for targetting this against [AIRFLOW-6976](https://issues.apache.org/jira/browse/AIRFLOW-6976) and not a new jira is around release-automation: if we have separate Jiras for a "related" fix then it is possible to miss this change if the first PR was backported. By having two commits target a single jira issue our existing release scripts (`./dev/airflow-jira`) will show both commits need to be pulled in.

I haven't noticed that before, thanks Ash for the clarification

**Comment by zhongjiajie** [comment]
Travis failed due to job name mistake and fix in https://github.com/apache/airflow/pull/7668, I rebase on master on restart the failing test.

**Comment by zhongjiajie** [comment]
Travis happy, could you please take a look? @ashb @potiuk @nuclearpinguin

**Comment by zhongjiajie** [comment]
PTAL @ashb @kaxil @potiuk @mik-laj @turbaszek @feluelle

File: UPDATING.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ https://developers.google.com/style/inclusive-documentation
 
 ### Deprecating ignore_first_depends_on_past on backfill command and default it to True
 
-When doing backfill with `depends_on_past` dags, users will need to pass `ignore_first_depends_on_past`.
+When doing backfill with `--depends-on-past` dags, users will need to pass `--ignore-first-depends-on-past`.
```
Comment (feluelle) [inline UPDATING.md]:
```suggestion
When doing backfill with `depends_on_past` dags, users will need to pass `--ignore-first-depends-on-past`.
```

Comment (feluelle) [reply UPDATING.md]: `depends_on_past` is not a cli option. It is an arg you can pass to tasks (`BaseOperator`)

**Review by feluelle — CHANGES_REQUESTED** [review]

File: UPDATING.md (hunk 1 above)
Comment (zhongjiajie) [reply UPDATING.md]: Oh, yeah a mistake

**Comment by zhongjiajie** [comment]
Fix UPDATING.md

**Review by kaxil — APPROVED** [review]

**Comment by zhongjiajie** [comment]
Thanks everyone

**Pull request merged**